src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -326,21 +347,29 @@ class ResourceService(
     */
   private def createOrUpdatePolicy(policyIdentity: FullyQualifiedPolicyId, policy: ValidatableAccessPolicy, samRequestContext: SamRequestContext): IO[AccessPolicy] = {
     val workbenchSubjects = policy.emailsToSubjects.values.flatten.toSet
-    accessPolicyDAO.loadPolicy(policyIdentity, samRequestContext).flatMap {
-      case None => createPolicy(policyIdentity, workbenchSubjects, generateGroupEmail(), policy.roles, policy.actions, policy.descendantPermissions, samRequestContext)
-      case Some(existingAccessPolicy) =>
-        val newAccessPolicy = AccessPolicy(policyIdentity, workbenchSubjects, existingAccessPolicy.email, policy.roles, policy.actions, policy.descendantPermissions, existingAccessPolicy.public)
-        if (newAccessPolicy == existingAccessPolicy) {
-          // short cut if access policy is unchanged
-          IO.pure(newAccessPolicy)
-        } else {
+    accessPolicyDAO.listAccessPolicies(policyIdentity.resource, samRequestContext).flatMap { originalPolicies =>
+      originalPolicies.find(_.id == policyIdentity) match {
+        case None =>
           for {
-            result <- accessPolicyDAO.overwritePolicy(newAccessPolicy, samRequestContext)
-            _ <- fireGroupUpdateNotification(policyIdentity, samRequestContext)
+            result <- createPolicy(policyIdentity, workbenchSubjects, generateGroupEmail(), policy.roles, policy.actions, policy.descendantPermissions, samRequestContext)
+            _ <- onPolicyUpdate(policyIdentity, originalPolicies, samRequestContext)
           } yield {
             result
           }
-        }
+        case Some(existingAccessPolicy) =>
+          val newAccessPolicy = AccessPolicy(policyIdentity, workbenchSubjects, existingAccessPolicy.email, policy.roles, policy.actions, policy.descendantPermissions, existingAccessPolicy.public)
+          if (newAccessPolicy == existingAccessPolicy) {
+            // short cut if access policy is unchanged
+            IO.pure(newAccessPolicy)
+          } else {
+            for {
+              result <- accessPolicyDAO.overwritePolicy(newAccessPolicy, samRequestContext)
```

Comment:
is this `result` different from `newAccessPolicy`?

Reply:
probably not